test/swap.js

```diff
@@ -3,12 +3,18 @@ var eq = require('./shared/eq.js');
 
 var list = ['a', 'b', 'c', 'd', 'e', 'f'];
 
+var a = ['a', 'A'];
+var b = ['b', 'B'];
+var c = ['c', 'C'];
+var d = ['d', 'D'];
+
 describe('swap', function() {
   it('swaps an element from one index to the other', function() {
     eq(R.swap(0, 1, list), ['b', 'a', 'c', 'd', 'e', 'f']);
     eq(R.swap(2, 1, list), ['a', 'c', 'b', 'd', 'e', 'f']);
     eq(R.swap(-1, 0, list), ['f', 'b', 'c', 'd', 'e', 'a']);
     eq(R.swap(4, 1, list), ['a', 'e', 'c', 'd', 'b', 'f']);
+    eq(R.swap(0, -1, [a, b, c, d]), [d, b, c, a]);
```

Comment:
Could we move this to its own `it`? Would be more clear that we are testing arrays.
```
it('works with lists of arrays', function() {
  eq(R.swap(0, -1, [['a', 'A'], ['b', 'B']]), [['b', 'B'], ['a', 'A']]);
});
```
or something like that